pkg/api/port_map.go

```diff
@@ -0,0 +1,36 @@
+package api
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+type PortMapParams struct {
+	SourceHost       string            `json:"source_host,omitempty"`
+	TargetHost       string            `json:"target_host,omitempty"`
+	TcpPorts         map[uint16]uint16 `json:"tcp_ports,omitempty"`
+	ExcludedTcpPorts []uint16          `json:"excluded_tcp_ports,omitempty"`
+	DefaultPortMap   bool              `json:"default_port_map,omitempty"`
+}
+
+func PortMap(request *http.Request) (interface{}, error) {
+	var (
+		portMapParams PortMapParams
+		data          interface{}
+	)
+
+	if err := json.NewDecoder(request.Body).Decode(&portMapParams); err != nil {
+		return nil, err
+	}
+
+	// place for executor
+	// Executor.execute // or something
+	output_mock := []byte("{\"foo\": \"bar\"}")
+	// return result to client
```

Comment:
Now we have the `executor` package that can be used here (instead of the mocking).